cluster.py

```diff
@@ -16,24 +16,32 @@
 tmp_dir = os.path.join(args.outdir, 'tmp')
 
 if args.lsf_index[0] == '\\':
-    args.lsf_index = args.lsf_index[1:] #this is for fixing a weird behaviour: the lsf_index comes with a "\" as the first character. The "\" is usefull to pass the parameter, but need to be deleted here. 
+    # this is for fixing a weird behaviour: the lsf_index comes with a "\" as the first character. The "\" is usefull to pass the parameter, but need to be deleted here.
+    args.lsf_index = args.lsf_index[1:]
 
 well_id = 'WE00' + '{:03d}'.format(int(args.lsf_index))
 analysis_id = args.channels + '-' + args.loops + '-' + well_id
 
-# Check if video for well id exists before producting data.
-if not io_operations.well_video_exists(args.indir, args.channels, args.loops, well_id):
+# check if there is a croppedRawTiff folder  ( I don´t know how to solve more elegantly, as the 'args = setup.parse_arguments()' above get the indir originally passed as argument, even if we change it later)
+subdir_list = os.listdir(args.indir)
+for n in subdir_list:
+    if 'croppedRAWTiff' in n:
+        args.indir = os.path.join(args.indir, 'croppedRAWTiff/', '')
+
+# Check if video for well id exists before producting data. Also check in CroppedRAWTiff folder.
+if not (io_operations.well_video_exists(args.indir, args.channels, args.loops, well_id) or io_operations.well_video_exists(args.indir + 'croppedRAWTiff/', args.channels, args.loops, well_id)):
```

Comment:
Is it necessary to check both folders? The loop above should change args.indir to args.indir + 'croppedRAWTiff/ if it exists?

Reply:
The answer to the second question is: Yes because in setup.py there is a function to strip out the croppedRAWTiff from the path if it is given in -indir:
# Move up one folder if croppedRAWTiff was given. The experiment folder is above it.
    if os.path.basename(os.path.normpath(args.indir)) == "croppedRAWTiff":
        args.indir = os.path.dirname(os.path.normpath(args.indir))
Sorry, but I did not understand your first question. Can you ask in a different way?

Reply:
At line 32, because of the loop above, args.indir will already be pointin to the subfolder croppedRAWTiff/ (if it is present)

So i think the second condition of the if statement, in line 32, is redundant, isn't it?

Reply:
I would not say that, because the first conditional actually change the args.indir variable, while the second only check for files inside args.indir path.

Reply:
Path1 = "/experiment_id/"
Path2 = "/experiment_id/croppedRAWTiff"

Path1 will be set from "experiment_id" to "/experiment_id/croppedRAWTiff" by the loop (lines 26-29)
So Path1 will be equal to Path2 at line 32.

Second conditional will check for "experiment_id/croppedRAWTiff/croppedRAWTiff" for both paths. (line 32)
I don't think this is ever the case.

Reply:
I see your point now and i thinK you are right. We should remove: "or io_operations.well_video_exists(args.indir + 'croppedRAWTiff/', args.channels, args.loops, well_id)"